components/director/internal/domain/token/resolver.go

```diff
@@ -0,0 +1,72 @@
+package token
+
+import (
+	"context"
+
+	"github.com/kyma-incubator/compass/components/director/internal/model"
+	"github.com/kyma-incubator/compass/components/director/internal/persistence"
+	"github.com/kyma-incubator/compass/components/director/pkg/graphql"
+	"github.com/pkg/errors"
+)
+
+//go:generate mockery -name=TokenService -output=automock -outpkg=automock -case=underscore
+type TokenService interface {
+	GenerateOneTimeToken(ctx context.Context, runtimeID string, tokenType model.SystemAuthReferenceObjectType) (model.OneTimeToken, error)
+}
+
+//go:generate mockery -name=TokenConverter -output=automock -outpkg=automock -case=underscore
+type TokenConverter interface {
+	ToGraphQL(model model.OneTimeToken) graphql.OneTimeToken
+}
+
+type Resolver struct {
+	transact persistence.Transactioner
+	svc      TokenService
+	conv     TokenConverter
+}
+
+func NewTokenResolver(transact persistence.Transactioner, svc TokenService, conv TokenConverter) *Resolver {
+	return &Resolver{transact: transact, svc: svc, conv: conv}
+}
+
+func (r *Resolver) GenerateOneTimeTokenForRuntime(ctx context.Context, id string) (*graphql.OneTimeToken, error) {
+	tx, err := r.transact.Begin()
```

Comment:
to check: how do we ensure that I cannot generate token for application for other tenant?

Reply:
https://github.com/kyma-incubator/compass/blob/6753554723fc903e97d5d0124b2864fbe621e304/components/schema-migrator/migrations/201909241337_add_system_auths.up.sql#L5
This constrain will prevent such action.